source/common/network/address_impl.cc

```
@@ -244,26 +244,44 @@ int Ipv6Instance::socket(SocketType type) const {
   return fd;
 }
 
-PipeInstance::PipeInstance(const sockaddr_un* address) : InstanceBase(Type::Pipe) {
+PipeInstance::PipeInstance(const sockaddr_un* address, socklen_t ss_len)
+    : InstanceBase(Type::Pipe) {
   if (address->sun_path[0] == '\0') {
-    throw EnvoyException("Abstract AF_UNIX sockets not supported.");
+    RELEASE_ASSERT(ss_len >= offsetof(struct sockaddr_un, sun_path) + 1);
+    abstract_namespace_ = true;
```

If this is not-linux, please throw an exception